cardano-node/test/Test/Cardano/Node/Chairman.hs

```diff
@@ -8,43 +9,69 @@ module Test.Cardano.Node.Chairman
 import           Cardano.Prelude
 import           Hedgehog (Property, discover)
 
+import qualified Data.Time.Clock as DTC
+import qualified Data.Time.Clock.POSIX as DTC
 import qualified Hedgehog as H
-import qualified System.Directory as IO
 import qualified System.IO as IO
 import qualified Test.Common.Base as H
 import qualified Test.Common.Process as H
 
 prop_spawnOneNode :: Property
 prop_spawnOneNode = H.propertyOnce . H.workspace "temp/chairman" $ \tempDir -> do
-  let dbDir = tempDir <> "/db/node-2"
-  let socketDir = tempDir <> "/socket"
+  base <- H.noteShowM H.getProjectBase
+  baseConfig <- H.noteShow $ base <> "/configuration/chairman/defaults/simpleview"
+  currentTime <- H.noteShowIO DTC.getCurrentTime
+  startTime <- H.noteShow $ DTC.addUTCTime 60 currentTime -- 60 seconds into the future
 
-  H.createDirectoryIfMissing dbDir
-  H.createDirectoryIfMissing socketDir
+  -- Generate keys
+  void $ H.execCli
+    [ "genesis"
+    , "--genesis-output-dir", tempDir <> "/genesis"
+    , "--start-time", show @Int64 (floor (DTC.utcTimeToPOSIXSeconds startTime))
+    , "--protocol-parameters-file", base <> "/scripts/protocol-params.json"
+    , "--k", "2160"
+    , "--protocol-magic", "459045235"
+    , "--n-poor-addresses", "128"
+    , "--n-delegate-addresses", "7"
+    , "--total-balance", "8000000000000000"
+    , "--avvm-entry-count", "128"
+    , "--avvm-entry-balance", "10000000000000"
+    , "--delegate-share", "0.9"
+    , "--real-pbft"
+    , "--secret-seed", "2718281828"
+    ]
 
-  base <- H.getProjectBase
+  -- Launch cluster of three nodes
+  procResults <- forM [0..2] $ \i -> do
+    si <- H.noteShow $ show @Int i
+    dbDir <- H.noteShow $ tempDir <> "/db/node-" <> si
+    socketDir <- H.noteShow $ tempDir <> "/socket"
 
-  dirContents <- liftIO $ IO.listDirectory base
+    H.createDirectoryIfMissing dbDir
```

Comment:
Is this not the same function as `createDirectoryIfMissing` ?

Reply:
It is exactly the same as `evalM . liftIO $ createDirectoryIfMissing`.

I've found the additional `evalM` is easily forgotten.  If forgotten then if/when an exception is thrown, you miss out on the nice output.